tests/rbd/test_rbd_pwl_cache_cluster_operation.py

```diff
@@ -0,0 +1,195 @@
+from ceph.rbd.workflows.cluster_operations import (
+    check_health,
+    operation,
+    osd_remove_and_add_back,
+    restart_ceph_target,
+    restart_osd,
+)
+from tests.rbd.rbd_peristent_writeback_cache import (
+    PersistentWriteAheadLog,
+    fio_ready,
+    get_entity_level,
+    kill_fio,
+)
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+
+def verify_cache(pwl, config, cache_client, image_spec):
+    """Method to verify rbd persistant cache test.
+
+    Args:
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict): The test configuration.
+        cache_client (Node): The client node for cache testing.
+        image_spec (str): The image specification for testing.
+    """
+    log.info("Writing data to the image using FIO jobs and checking cache status")
+    with parallel() as p:
+        p.spawn(
+            run_fio, **fio_ready(config, cache_client, verify="crc32", verify_fatal=1)
+        )
+        p.spawn(
+            pwl.check_cache_file_exists,
+            image_spec,
+            config["fio"].get("runtime", 30),
+            **config,
+        )
+
+
+def run(ceph_cluster, **kw):
+    """RBD Persistent write back cache with cluster operations.
+
+    Args:
+        ceph_cluster: ceph cluster object
+        **kw: test parameters
+    """
+    log.info(
+        "Running test - CEPH-83574891 - Verify ceph cluster operations "
+        "with persistent cache enabled in SSD mode with IO in-progress"
+    )
+    config = kw.get("config")
+    rbd_obj = initial_rbd_config(**kw)["rbd_reppool"]
+    pool = config["rep_pool_config"]["pool"]
+    image_spec = (
+        f"{config['rep_pool_config']['pool']}/{config['rep_pool_config']['image']}"
+    )
+    cache_client = ceph_cluster.get_nodes(role="client")[0]
+    admin_node = ceph_cluster.get_nodes(role="_admin")[0]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mon_obj = MonitorWorkflows(node=cephadm)
+    pwl = PersistentWriteAheadLog(rbd_obj, cache_client, config.get("drive"))
+    config_level, entity = get_entity_level(config)
+
+    try:
+        # Configute cache client
+        pwl.configure_cache_client()
+
+        # Configure PWL
+        pwl.configure_pwl_cache(
+            config["rbd_persistent_cache_mode"],
+            config_level,
+            entity,
+            config["cache_file_size"],
+        )
+
+        # Test to verify persistent write back cache test
+        verify_cache(pwl, config, cache_client, image_spec)
+
+        log.info("Test to kill FIO process and restarting FIO to check cache status")
+        with parallel() as p:
+            p.spawn(run_fio, **fio_ready(config, cache_client))
+            p.spawn(kill_fio, pwl.rbd)
+
+        verify_cache(pwl, config, cache_client, image_spec)
```

Comment:
verify_cache should also be part of p.spawn above.

Reply:
no here after killing fio only we are rechecking `verify_cache` with new fio hence it's a separate test not in parallel,